### PR TITLE
docs(guides): updates examples of glob patterns supported in dagger

### PR DESCRIPTION
**Pull request by JammUtkarsh**
Solves: #5070

This PR updates the existing documentaion to show support for support of glob patterns in dagger SDKs.

### CODE REVIEW

File: docs/current/guides/421437-work-with-host-filesystem.md
Comment (vikram-dagger) [inline docs/current/guides/421437-work-with-host-filesystem.md:138]: I would prefer not to delete the existing snippets. Can we restore this snippet to its original form and place the new glob pattern example as a new example in the same section, after this one?

File: docs/current/guides/snippets/work-with-host-filesystem/glob-pattern/index.mts
Comment (vikram-dagger) [inline docs/current/guides/snippets/work-with-host-filesystem/glob-pattern/index.mts:32]: This is linked to the ongoing discussion in https://discord.com/channels/707636530424053791/1145698713331834990/1146770191120875580 and the PR https://github.com/dagger/dagger/pull/5725/ which makes changes to the existing examples. I recommend waiting until that discussion is closed and PR is merged and then updating these examples.

File: docs/current/guides/snippets/work-with-host-filesystem/glob-pattern/main.go
Comment (vikram-dagger) [inline docs/current/guides/snippets/work-with-host-filesystem/glob-pattern/main.go:28]: See https://github.com/dagger/dagger/pull/5719/files#r1312131419

File: docs/current/guides/snippets/work-with-host-filesystem/glob-pattern/main.py
Comment (vikram-dagger) [inline docs/current/guides/snippets/work-with-host-filesystem/glob-pattern/main.py:34]: See https://github.com/dagger/dagger/pull/5719/files#r1312131419

**Comment by TomChv** [comment]
@vikram-dagger Is this PR ready to be merged? @JammUtkarsh Applied changes you ask

**Comment by JammUtkarsh** [comment]
I am sorry, I have been a little busy with my school.
I'll push the changes by this weekend.

**Review by TomChv — APPROVED** [review]
I rebased the PR, code and explanations LGTM, I let @vikram-dagger give the final approval

**Review by vikram-dagger — APPROVED** [review]

**Pull request merged**
